src/session_pool/session.js

```diff
@@ -0,0 +1,175 @@
+import { cryptoRandomObjectId } from 'apify-shared/utilities';
+
+import { checkParamOrThrow } from 'apify-client/build/utils';
+import EVENTS from './events';
+
+
+/**
+ *  Class aggregating data for session.
+ *  Sessions are used to store information such as cookies and can be used for generating fingerprints and proxy sessions.
+ *  You can imagine each session as a specific user, with its own cookies, IP (via proxy) and potentially a unique browser fingerprint.
+ *  Session internal state can be enriched with custom user data for example some authorization tokens and specific headers in general.
+ */
+export class Session {
+    /**
+     * Session configuration.
+     * @param options
+     * @param options.id {String} - Id of session used for generating fingerprints. It is used as proxy session name.
+     * @param options.maxAgeSecs {Number} - Number of seconds after which the session is considered as expired.
+     * @param options.userData {Object} - Object where custom user data can be stored. For example custom headers.
+     * @param options.maxErrorScore {number} - Maximum number of marking session as blocked usage.
```

Comment:
There is no info about the scale of these "error score" numbers. Is 5 large or not? How about 0.2? I don't know.... Basically you don't mention anywhere that error score starts at 0, that `markBad` increments it by 1 and `markGood` decrements it by `errorScoreDecrement`. So please either explain it better in comments or choose variable names that will make it clear from the context.